src/main/java/anab/Person/Trainer.java

```diff
@@ -5,39 +5,62 @@
 
 public class Trainer extends Person {
     private String specialization;
-    private int experience;
+    private int experienceYears;
     private Map<String, Integer> feedbackMarks = new HashMap<>();
 
-    public Trainer(String firstName, String lastName, String gender, int age, String emailAddress, boolean isEmailRestricted, String specialization, int experience, Map<String, Integer> feedbackMarks) {
+    //constructor explicit -mosteneste Person=> super(param1, param2, param3,...)
+    public Trainer(String firstName, String lastName, String gender, int age, String emailAddress, boolean isEmailRestricted, String specialization, int experienceYears, Map<String, Integer> feedbackMarks) {
         super(firstName, lastName, gender, age, emailAddress, isEmailRestricted);
         this.specialization = specialization;
-        this.experience = experience;
+        this.experienceYears = experienceYears;
         this.feedbackMarks = feedbackMarks;
     }
 
+    // see the specialization of a trainer
     public String getSpecialization() {
-        return specialization;
+        return this.specialization;
     }
 
-    public String setSpecialization(String specialization) {
+    //add specialization of a trainer
+    public void setSpecialization(String specialization) {
         this.specialization = specialization;
-        return specialization;
     }
 
+    //see the years of experience of a trainer
     public int getExperience() {
-        return experience;
-    }
-    public int addExperience(int years) {
-        this.experience=years;
-        return years;
-    }
-    public void setExperience(String specialization, int years) {
-        this.experience = years;
-        if(years!=0 && years>0){
-            experience=addExperience(years);
-            System.out.println(years+ " years of experience on "+ specialization);
-        }else{
+        return this.experienceYears;
+    }
+
+    //Add years of experience for a trainer
+    public void addExperience(String specialization, int experienceYears) {
+        this.specialization = specialization;
+        this.experienceYears = experienceYears;
+
+    }
+
+    public void setExperienceYears(int years, int existingExp) {
+        if (years != 0 && years > 0 && years >= existingExp) {
+            years = getExperience() + years;
+            System.out.println(years + " years of experience on " + this.specialization);
+        } else {
             System.out.println("Add valid value for years of experience= ");
         }
     }
-}
+
+    public void addFeedbackMark(int feedback, String name) {
+        if (feedback <= 0 || feedback > 10) {
+            if (name == " ") {
```

Comment:
keep in mind that the logic inside this method should be changed; a part of the logic will be moved in the overloaded method

Reply:
Splitted into two methods: 

- base method **addFeedbackMark** with 1 param 

- overload method **addFeedbackMark** having 2 params defined